core/sdk.go

```diff
@@ -35,66 +37,188 @@ type PolicyResult struct {
 // Warning: This interface is experimental, and it could change with breaking also in rond patches.
 // Does not use outside this repository until it is not ready.
 type SDK interface {
-	// Warning: this method will be removed in the near future. Do not use it outside Rond.
-	Metrics() metrics.Metrics
-
 	FindEvaluator(logger *logrus.Entry, method, path string) (SDKEvaluator, error)
-	EvaluatorFromConfig(logger *logrus.Entry, config openapi.RondConfig) SDKEvaluator
 }
 
 // Warning: This interface is experimental, and it could change with breaking also in rond patches.
 // Do not use outside this repository until it is not ready.
 type SDKEvaluator interface {
 	Config() openapi.RondConfig
-	PartialResultsEvaluators() PartialResultsEvaluators
+
+	EvaluateRequestPolicy(ctx context.Context, req RondInput, userInfo types.User) (PolicyResult, error)
+	EvaluateResponsePolicy(ctx context.Context, rondInput RondInput, userInfo types.User, decodedBody any) ([]byte, error)
 }
 
 type evaluator struct {
 	rond       rondImpl
 	logger     *logrus.Entry
 	rondConfig openapi.RondConfig
+
+	routeInfo openapi.RouterInfo
+}
+
+func (e evaluator) metrics() metrics.Metrics {
+	return e.rond.metrics
+}
+
+func (e evaluator) partialResultEvaluators() PartialResultsEvaluators {
+	return e.rond.partialResultEvaluators
 }
 
 func (e evaluator) Config() openapi.RondConfig {
 	return e.rondConfig
 }
 
-func (e evaluator) PartialResultsEvaluators() PartialResultsEvaluators {
-	return e.rond.evaluator
+func (e evaluator) EvaluateRequestPolicy(ctx context.Context, req RondInput, userInfo types.User) (PolicyResult, error) {
```

Comment:
I'd return a pointer to `PolicyResult` what's the advantage of copying it on return? 🤔

Reply:
That it cannot be nil, and it default to the correct values. Also, the struct is very small